src/routes/api/urls.go

```diff
@@ -21,24 +22,64 @@ func UrlsRoute() func(router fiber.Router) {
 	urlsController = controllers.CreateUrlsController()
 
 	return func(router fiber.Router) {
-		router.Get("/", getAllUrls)
+		router.Get("/", security.OptionalJwtAuthMiddleware, getAllUrls)
 		router.Post("/", security.MandatoryJwtAuthMiddleware, createRandomUrl)
 		router.Put("/:shortcode", security.MandatoryJwtAuthMiddleware, createSpecificUrl)
 	}
 }
 
 // getAllUrls
 //
-//	@Summary		Get all urls
-//	@Description	Get all urls
+//	@Summary		Get all urls (for users or admins)
+//	@Description	Get all urls for the current user or all URLs for admins
 //	@Tags			urls
 //	@Accept			json
 //	@Produce		json
-//	@Success		200	{string}	string	"GetAllUsers"
+//	@Param			X-API-Key	header		string	false	"Admin API Key (optional)"
+//	@Param			userid		query		string	false	"Filter URLs by user ID (admin only)"
+//	@Success		200			{array}		dtos.UrlResponse
+//	@Failure		400			{object}	dtos.ErrorResponse	"Invalid user ID"
+//	@Failure		401			{object}	dtos.ErrorResponse	"Unauthorized"
+//	@Failure		500			{object}	dtos.ErrorResponse	"Failed to fetch URLs"
 //	@Router			/urls [get]
-//	@security		BearerToken
+//	@Security		BearerToken
+//	@Security		ApiKeyAuth
 func getAllUrls(ctx *fiber.Ctx) error {
-	return ctx.SendString("GetAllUsers")
+	apiKey := ctx.Get("X-API-Key")
```

Comment:
can we fail fast if neither JWT nor API Key is given?

Reply:
Sure, I'll do the same